app/src/main/java/com/door43/translationstudio/newui/translate/ChunkModeAdapter.java

```diff
@@ -993,5 +1010,217 @@ public boolean isFrame() {
+                    }
+                } else { // enable highlighting
+                    renderedTargetBody = null; // remove rendered text so will be re-rendered with new highlighting
+                }
+                mHighlightTarget = enable;
+            } else { // source
+                if(!enable) { // disable highlighting
+                    if(mHighlightSource) {
+                        renderedSourceBody = null; // remove rendered text so will be re-rendered without highlighting
+                    }
+                } else { // enable highlighting
+                    renderedSourceBody = null; // remove rendered text so will be re-rendered with new highlighting
+                }
+                mHighlightSource = enable;
+            }
+        }
+
+        /**
+         * Loads the correct translation information into the item
+         * @param targetTranslation
+         * @param chapter
+         * @param frame
+         */
+        public void loadTranslations(SourceTranslation sourceTranslation, TargetTranslation targetTranslation, Chapter chapter, Frame frame, ChunkModeAdapter adapter) {
+            if(isChapterReference || isChapterTitle) {
+                ChapterTranslation chapterTranslation = targetTranslation.getChapterTranslation(chapter);
+                if (isChapterTitle) {
+                    renderedTargetBody = chapterTranslation.title;
+                    renderedSourceBody = chapter.title;
+                } else {
+                    renderedTargetBody = chapterTranslation.reference;
+                    renderedSourceBody = chapter.reference;
+                }
+            } else if(isProjectTitle) {
+                ProjectTranslation projectTranslation = targetTranslation.getProjectTranslation();
+                renderedTargetBody = projectTranslation.getTitle();
+                renderedSourceBody = sourceTranslation.getProjectTitle();
+            } else {
+                FrameTranslation frameTranslation = targetTranslation.getFrameTranslation(frame);
+                TranslationFormat targetFormat =  targetTranslation.getFormat();
+                boolean targetSearch = adapter.isTargetSearch();
+                renderedSourceBody = adapter.renderText(frame.body, frame.getFormat(),!targetSearch);
+                renderedTargetBody = adapter.renderText(frameTranslation.body, targetFormat, targetSearch);
+            }
+        }
+    }
+
+    /**
+     * Loads a frame from the index and caches it
+     * @param chapterSlug
+     * @param frameSlug
+     * @return
+     */
+    private Frame loadFrame(String chapterSlug, String frameSlug) {
+        String complexSlug = chapterSlug + "-" + frameSlug;
+        Frame frame = mLibrary.getFrame(mSourceTranslation, chapterSlug, frameSlug);
+        return frame;
+    }
+
+    /**
+     * clear the cards displayed since we have new search string
+     * @return
+     */
+    private void clearCardDisplay() {
```

Comment:
This is only used once and it's only 2 lines of code so this would be better suited inline. If you want to add clarity of purpose you can add a one line comment like
```
# clear card display for search
mFilteredItems = new ListItem[0];
notifyDataSetChanged();
```